diff -urN hdf5-1.14.4.3-orig/src/H5win32defs.h hdf5-1.14.4.3/src/H5win32defs.h
--- hdf5-1.14.4.3-orig/src/H5win32defs.h	2024-05-23 02:20:53.000000000 +0800
+++ hdf5-1.14.4.3/src/H5win32defs.h	2025-11-02 11:15:49.720537000 +0800
@@ -33,11 +33,13 @@
 typedef __int64         h5_stat_size_t;
 
 #ifdef H5_HAVE_VISUAL_STUDIO
+#ifndef _TIMEZONE_DEFINED
 struct timezone {
     int tz_minuteswest;
     int tz_dsttime;
 };
 #endif
+#endif
 
 #define HDcreat(S, M)        Wopen(S, O_CREAT | O_TRUNC | O_RDWR, M)
 #define HDflock(F, L)        Wflock(F, L)
