diff -urN global-6.6.14-orig/global/literal.c global-6.6.14/global/literal.c
--- global-6.6.14-orig/global/literal.c	2024-12-11 14:44:34.000000000 +0800
+++ global-6.6.14/global/literal.c	2025-11-16 11:31:45.239671500 +0800
@@ -29,7 +29,9 @@
 #endif
 #include <stdio.h>
 #include <ctype.h>
+#ifdef HAVE_SYS_PARAM_H
 #include <sys/param.h>
+#endif
 #include <sys/stat.h>
 #include <errno.h>
 #ifdef STDC_HEADERS
diff -urN global-6.6.14-orig/htags/htags.c global-6.6.14/htags/htags.c
--- global-6.6.14-orig/htags/htags.c	2024-12-11 14:44:34.000000000 +0800
+++ global-6.6.14/htags/htags.c	2025-11-16 11:48:06.294052300 +0800
@@ -38,7 +38,9 @@
 #include <signal.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#ifdef HAVE_SYS_PARAM_H
 #include <sys/param.h>
+#endif
 #include <errno.h>
 
 #include "getopt.h"
diff -urN global-6.6.14-orig/htags-refkit/htags_path2url.c global-6.6.14/htags-refkit/htags_path2url.c
--- global-6.6.14-orig/htags-refkit/htags_path2url.c	2024-12-11 14:44:34.000000000 +0800
+++ global-6.6.14/htags-refkit/htags_path2url.c	2025-11-16 11:31:45.292283200 +0800
@@ -10,7 +10,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_SYS_PARAM_H
 #include <sys/param.h>
+#endif
 #include <sys/stat.h>
 
 #if defined(_WIN32) || defined(__DJGPP__)
diff -urN global-6.6.14-orig/libdb/sqlite3.c global-6.6.14/libdb/sqlite3.c
--- global-6.6.14-orig/libdb/sqlite3.c	2024-12-11 14:44:34.000000000 +0800
+++ global-6.6.14/libdb/sqlite3.c	2025-11-16 11:47:46.014686600 +0800
@@ -24697,7 +24697,9 @@
 #  include <limits.h>
 # else
 #  include <sys/file.h>
+#ifdef HAVE_SYS_PARAM_H
 #  include <sys/param.h>
+#endif
 # endif
 #endif /* SQLITE_ENABLE_LOCKING_STYLE */
 
diff -urN global-6.6.14-orig/libglibc/regex.c global-6.6.14/libglibc/regex.c
--- global-6.6.14-orig/libglibc/regex.c	2024-12-11 14:44:34.000000000 +0800
+++ global-6.6.14/libglibc/regex.c	2025-11-16 11:31:45.332146500 +0800
@@ -297,11 +297,14 @@
 /* Make alloca work the best possible way.  */
 #  ifdef __GNUC__
 #   define alloca __builtin_alloca
-#  else /* not __GNUC__ */
+#  elif defined _MSC_VER
+#   include <malloc.h>
+#   define alloca _alloca
+#  else /* not __GNUC__ and _MSC_VER */
 #   if HAVE_ALLOCA_H
 #    include <alloca.h>
 #   endif /* HAVE_ALLOCA_H */
-#  endif /* not __GNUC__ */
+#  endif /* not __GNUC__ and _MSC_VER */
 
 # endif /* not alloca */
 
diff -urN global-6.6.14-orig/libltdl/Makefile.in global-6.6.14/libltdl/Makefile.in
--- global-6.6.14-orig/libltdl/Makefile.in	2024-12-11 14:44:42.000000000 +0800
+++ global-6.6.14/libltdl/Makefile.in	2025-11-16 11:31:45.332146500 +0800
@@ -299,7 +299,7 @@
 # -I$(srcdir) is needed for user that built libltdl with a sub-Automake
 # (not as a sub-package!) using 'nostdinc':
 AM_CPPFLAGS = -DLT_CONFIG_H='<$(LT_CONFIG_H)>' -DLTDL -I. -I$(srcdir) \
-	-Ilibltdl -I$(srcdir)/libltdl
+	-Ilibltdl -I$(srcdir)/libltdl -I$(top_srcdir)/libglibc
 AM_DEFAULT_VERBOSITY = @AM_DEFAULT_VERBOSITY@
 AR = @AR@
 AUTOCONF = @AUTOCONF@
diff -urN global-6.6.14-orig/libutil/gparam.h global-6.6.14/libutil/gparam.h
--- global-6.6.14-orig/libutil/gparam.h	2024-12-11 14:44:34.000000000 +0800
+++ global-6.6.14/libutil/gparam.h	2025-11-16 11:31:45.347768200 +0800
@@ -20,7 +20,7 @@
 
 #ifndef _GPARAM_H_
 #define _GPARAM_H_
-#ifndef __BORLANDC__
+#ifdef HAVE_SYS_PARAM_H
 #include <sys/param.h>
 #endif
 
diff -urN global-6.6.14-orig/libutil/Makefile.in global-6.6.14/libutil/Makefile.in
--- global-6.6.14-orig/libutil/Makefile.in	2024-12-11 14:44:42.000000000 +0800
+++ global-6.6.14/libutil/Makefile.in	2025-11-16 11:31:45.347768200 +0800
@@ -257,7 +257,8 @@
 	-DDATADIR='"$(datadir)"' \
 	-DLIBDIR='"$(libdir)"' \
 	-DLOCALSTATEDIR='"$(localstatedir)"' \
-	-DSYSCONFDIR='"$(sysconfdir)"'
+	-DSYSCONFDIR='"$(sysconfdir)"' \
+	-I$(top_srcdir)/libglibc
 
 AM_DEFAULT_VERBOSITY = @AM_DEFAULT_VERBOSITY@
 AR = @AR@
