diff -urN PostgreSQL-16.9-orig/src/port/getopt.c PostgreSQL-16.9/src/port/getopt.c
--- PostgreSQL-16.9-orig/src/port/getopt.c	2025-05-06 04:30:08.000000000 +0800
+++ PostgreSQL-16.9/src/port/getopt.c	2025-11-14 10:40:18.137916600 +0800
@@ -57,6 +57,7 @@
 #define BADARG	(int)':'
 #define EMSG	""
 
+#ifndef HAVE_GETOPT
 /*
  * getopt
  *	Parse argc/argv argument vector.
@@ -134,3 +135,4 @@
 	}
 	return optopt;				/* dump back option letter */
 }
+#endif
diff -urN PostgreSQL-16.9-orig/src/port/getopt_long.c PostgreSQL-16.9/src/port/getopt_long.c
--- PostgreSQL-16.9-orig/src/port/getopt_long.c	2025-05-06 04:30:08.000000000 +0800
+++ PostgreSQL-16.9/src/port/getopt_long.c	2025-11-14 10:40:35.568924200 +0800
@@ -43,6 +43,7 @@
 #define EMSG	""
 
 
+#ifndef HAVE_GETOPT_LONG
 /*
  * getopt_long
  *	Parse argc/argv argument vector, with long options.
@@ -216,3 +217,4 @@
 	}
 	return optopt;
 }
+#endif
diff -urN PostgreSQL-16.9-orig/src/tools/msvc/Solution.pm PostgreSQL-16.9/src/tools/msvc/Solution.pm
--- PostgreSQL-16.9-orig/src/tools/msvc/Solution.pm	2025-05-06 04:30:08.000000000 +0800
+++ PostgreSQL-16.9/src/tools/msvc/Solution.pm	2025-11-14 10:25:10.264064500 +0800
@@ -257,9 +257,9 @@
 		HAVE_GCC__SYNC_INT32_TAS => undef,
 		HAVE_GCC__SYNC_INT64_CAS => undef,
 		HAVE_GETIFADDRS => undef,
-		HAVE_GETOPT => undef,
-		HAVE_GETOPT_H => undef,
-		HAVE_GETOPT_LONG => undef,
+		HAVE_GETOPT => 1,
+		HAVE_GETOPT_H => 1,
+		HAVE_GETOPT_LONG => 1,
 		HAVE_GETPEEREID => undef,
 		HAVE_GETPEERUCRED => undef,
 		HAVE_GSSAPI_EXT_H => undef,
@@ -340,7 +340,7 @@
 		HAVE_STRLCPY => undef,
 		HAVE_STRNLEN => 1,
 		HAVE_STRSIGNAL => undef,
-		HAVE_STRUCT_OPTION => undef,
+		HAVE_STRUCT_OPTION => 1,
 		HAVE_STRUCT_SOCKADDR_SA_LEN => undef,
 		HAVE_STRUCT_TM_TM_ZONE => undef,
 		HAVE_SYNC_FILE_RANGE => undef,
@@ -1031,6 +1031,7 @@
 	{
 		$proj->AddIncludeDir($self->{options}->{nls} . '\include');
 		$proj->AddLibrary($self->{options}->{nls} . '\lib\libintl.lib');
+		$proj->AddLibrary($self->{options}->{nls} . '\lib\pcrt.lib');
 	}
 	if ($self->{options}->{gss})
 	{
@@ -1039,20 +1040,20 @@
 		if ($self->{platform} eq 'Win32')
 		{
 			$proj->AddLibrary(
-				$self->{options}->{gss} . '\lib\i386\krb5_32.lib');
+				$self->{options}->{gss} . '\lib\krb5_32.lib');
 			$proj->AddLibrary(
-				$self->{options}->{gss} . '\lib\i386\comerr32.lib');
+				$self->{options}->{gss} . '\lib\comerr32.lib');
 			$proj->AddLibrary(
-				$self->{options}->{gss} . '\lib\i386\gssapi32.lib');
+				$self->{options}->{gss} . '\lib\gssapi32.lib');
 		}
 		else
 		{
 			$proj->AddLibrary(
-				$self->{options}->{gss} . '\lib\amd64\krb5_64.lib');
+				$self->{options}->{gss} . '\lib\krb5_64.lib');
 			$proj->AddLibrary(
-				$self->{options}->{gss} . '\lib\amd64\comerr64.lib');
+				$self->{options}->{gss} . '\lib\comerr64.lib');
 			$proj->AddLibrary(
-				$self->{options}->{gss} . '\lib\amd64\gssapi64.lib');
+				$self->{options}->{gss} . '\lib\gssapi64.lib');
 		}
 	}
 	if ($self->{options}->{iconv})
@@ -1063,18 +1064,9 @@
 	if ($self->{options}->{icu})
 	{
 		$proj->AddIncludeDir($self->{options}->{icu} . '\include');
-		if ($self->{platform} eq 'Win32')
-		{
-			$proj->AddLibrary($self->{options}->{icu} . '\lib\icuin.lib');
-			$proj->AddLibrary($self->{options}->{icu} . '\lib\icuuc.lib');
-			$proj->AddLibrary($self->{options}->{icu} . '\lib\icudt.lib');
-		}
-		else
-		{
-			$proj->AddLibrary($self->{options}->{icu} . '\lib64\icuin.lib');
-			$proj->AddLibrary($self->{options}->{icu} . '\lib64\icuuc.lib');
-			$proj->AddLibrary($self->{options}->{icu} . '\lib64\icudt.lib');
-		}
+		$proj->AddLibrary($self->{options}->{icu} . '\lib\icuin.lib');
+		$proj->AddLibrary($self->{options}->{icu} . '\lib\icuuc.lib');
+		$proj->AddLibrary($self->{options}->{icu} . '\lib\icudt.lib');
 	}
 	if ($self->{options}->{xml})
 	{
