diff -urN speex-1.2.1-orig/libspeex/stack_alloc.h speex-1.2.1/libspeex/stack_alloc.h
--- speex-1.2.1-orig/libspeex/stack_alloc.h	2022-05-03 11:09:25.000000000 +0800
+++ speex-1.2.1/libspeex/stack_alloc.h	2025-11-17 11:42:07.538697500 +0800
@@ -38,6 +38,7 @@
 #ifdef USE_ALLOCA
 # ifdef _WIN32
 #  include <malloc.h>
+#  define alloca _alloca
 # else
 #  ifdef HAVE_ALLOCA_H
 #   include <alloca.h>
diff -urN speex-1.2.1-orig/libspeex/vorbis_psy.c speex-1.2.1/libspeex/vorbis_psy.c
--- speex-1.2.1-orig/libspeex/vorbis_psy.c	2018-09-08 06:20:02.000000000 +0800
+++ speex-1.2.1/libspeex/vorbis_psy.c	2025-11-17 11:42:04.439724100 +0800
@@ -45,6 +45,10 @@
 #include <math.h>
 #include <string.h>
 
+#if defined(_MSC_VER) && !defined(alloca)
+# define alloca _alloca
+#endif
+
 /* psychoacoustic setup ********************************************/
 
 static VorbisPsyInfo example_tuning = {
