diff -urN ncurses-6.5-orig/include/win32_curses.h ncurses-6.5/include/win32_curses.h
--- ncurses-6.5-orig/include/win32_curses.h	2021-06-18 05:26:02.000000000 +0800
+++ ncurses-6.5/include/win32_curses.h	2024-11-26 15:53:17.535982000 +0800
@@ -68,8 +68,10 @@
 #undef sleep
 #define sleep(n) Sleep((n) * 1000)
 
+#if !HAVE_CLOCK_GETTIME && !HAVE_GETTIMEOFDAY
 #undef gettimeofday
 #define gettimeofday(tv,tz) _nc_gettimeofday(tv,tz)
+#endif
 
 #endif /* _NC_WIN32_CURSES_H */
 #endif /* _WIN32||_WIN64 */
diff -urN ncurses-6.5-orig/ncurses/tinfo/lib_win32con.c ncurses-6.5/ncurses/tinfo/lib_win32con.c
--- ncurses-6.5-orig/ncurses/tinfo/lib_win32con.c	2023-08-06 04:44:38.000000000 +0800
+++ ncurses-6.5/ncurses/tinfo/lib_win32con.c	2024-11-26 15:53:17.551633000 +0800
@@ -36,6 +36,7 @@
  * TODO - GetMousePos(POINT * result) from ntconio.c
  */
 
+#define NEED_KEY_EVENT
 #include <curses.priv.h>
 
 MODULE_ID("$Id: lib_win32con.c,v 1.14 2023/08/05 20:44:38 tom Exp $")
