diff -urN scotch-7.0.6-orig/src/libscotch/common.h scotch-7.0.6/src/libscotch/common.h
--- scotch-7.0.6-orig/src/libscotch/common.h	2024-12-06 06:09:54.000000000 +0800
+++ scotch-7.0.6/src/libscotch/common.h	2025-11-16 09:18:55.288080100 +0800
@@ -103,7 +103,9 @@
 #define strcasecmp                  stricmp
 #endif /* ((defined _WIN32) && (! defined __MINGW32__)) */
 #define pipe(fd)                    _pipe (fd, 32768, O_BINARY)
+#ifndef HAVE_SLEEP
 #define sleep(s)                    Sleep (1000 * (s))
+#endif
 #endif /* COMMON_OS_WINDOWS */
 
 #if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
