diff -urN libvideogfx-1.0.8-orig/examples/Makefile.in libvideogfx-1.0.8/examples/Makefile.in
--- libvideogfx-1.0.8-orig/examples/Makefile.in	2013-08-27 01:15:12.000000000 +0800
+++ libvideogfx-1.0.8/examples/Makefile.in	2025-11-16 03:04:07.208219100 +0800
@@ -386,129 +386,129 @@
 test_libvideogfx_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 test_libvideogfx_CFLAGS = $(X_CFLAGS)
 test_libvideogfx_LDFLAGS = -g $(X_LDFLAGS)
-test_libvideogfx_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+test_libvideogfx_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 test_libvideogfx_SOURCES = test-libvideogfx.cc
 dynarray_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 dynarray_CFLAGS = $(X_CFLAGS)
 dynarray_LDFLAGS = -g $(X_LDFLAGS)
-dynarray_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+dynarray_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 dynarray_SOURCES = dynarray.cc
 symmatrix_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 symmatrix_CFLAGS = $(X_CFLAGS)
 symmatrix_LDFLAGS = -g $(X_LDFLAGS)
-symmatrix_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+symmatrix_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 symmatrix_SOURCES = symmatrix.cc
 bitstream_output_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 bitstream_output_CFLAGS = $(X_CFLAGS)
 bitstream_output_LDFLAGS = -g $(X_LDFLAGS)
-bitstream_output_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+bitstream_output_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 bitstream_output_SOURCES = bitstream_output.cc
 
 #smartpointer_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 #smartpointer_CFLAGS = $(X_CFLAGS)
 #smartpointer_LDFLAGS = -g $(X_LDFLAGS)
-#smartpointer_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+#smartpointer_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 #smartpointer_SOURCES = smartpointer.cc
 
 #ppm2avi_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 #ppm2avi_CFLAGS = $(X_CFLAGS)
 #ppm2avi_LDFLAGS = -g $(X_LDFLAGS)
-#ppm2avi_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+#ppm2avi_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 #ppm2avi_SOURCES = ppm2avi.cc
 cputest_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 cputest_CFLAGS = $(X_CFLAGS)
 cputest_LDFLAGS = -g $(X_LDFLAGS)
-cputest_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+cputest_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 cputest_SOURCES = cputest.cc
 pixmaptest_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 pixmaptest_CFLAGS = $(X_CFLAGS)
 pixmaptest_LDFLAGS = -g $(X_LDFLAGS)
-pixmaptest_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+pixmaptest_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 pixmaptest_SOURCES = pixmaptest.cc
 filtertest_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 filtertest_CFLAGS = $(X_CFLAGS)
 filtertest_LDFLAGS = -g $(X_LDFLAGS)
-filtertest_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+filtertest_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 filtertest_SOURCES = filtertest.cc
 ppmflip_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 ppmflip_CFLAGS = $(X_CFLAGS)
 ppmflip_LDFLAGS = -g $(X_LDFLAGS)
-ppmflip_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+ppmflip_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 ppmflip_SOURCES = ppmflip.cc
 chromatest_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 chromatest_CFLAGS = $(X_CFLAGS)
 chromatest_LDFLAGS = -g $(X_LDFLAGS)
-chromatest_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+chromatest_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 chromatest_SOURCES = chromatest.cc
 @HAVE_X11_TRUE@x11test_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@x11test_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@x11test_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@x11test_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@x11test_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@x11test_SOURCES = x11test.cc
 @HAVE_X11_TRUE@x11paint_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@x11paint_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@x11paint_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@x11paint_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@x11paint_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@x11paint_SOURCES = x11paint.cc
 @HAVE_X11_TRUE@viewfile_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@viewfile_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@viewfile_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@viewfile_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@viewfile_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@viewfile_SOURCES = viewfile.cc
 @HAVE_X11_TRUE@persptransform_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@persptransform_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@persptransform_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@persptransform_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@persptransform_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@persptransform_SOURCES = persptransform.cc
 @HAVE_X11_TRUE@convfile_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@convfile_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@convfile_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@convfile_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@convfile_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@convfile_SOURCES = convfile.cc
 @HAVE_X11_TRUE@median_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@median_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@median_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@median_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@median_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@median_SOURCES = median.cc
 @HAVE_X11_TRUE@filedisplay_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@filedisplay_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@filedisplay_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@filedisplay_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@filedisplay_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@filedisplay_SOURCES = filedisplay.cc
 @HAVE_X11_TRUE@chromaspaces_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@chromaspaces_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@chromaspaces_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@chromaspaces_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@chromaspaces_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@chromaspaces_SOURCES = chromaspaces.cc
 @HAVE_X11_TRUE@imagetest_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@imagetest_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@imagetest_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@imagetest_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@imagetest_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@imagetest_SOURCES = imagetest.cc
 @HAVE_X11_TRUE@drawingtest_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@drawingtest_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@drawingtest_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@drawingtest_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@drawingtest_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@drawingtest_SOURCES = drawingtest.cc
 @HAVE_X11_TRUE@copytest_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@copytest_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@copytest_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@copytest_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@copytest_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@copytest_SOURCES = copytest.cc
 audioout_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 audioout_CFLAGS = $(X_CFLAGS)
 audioout_LDFLAGS = -g $(X_LDFLAGS)
-audioout_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+audioout_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 audioout_SOURCES = audioout.cc
 @HAVE_X11_TRUE@@LINUX_TRUE@v4ltest_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@@LINUX_TRUE@v4ltest_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@@LINUX_TRUE@v4ltest_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@@LINUX_TRUE@v4ltest_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@@LINUX_TRUE@v4ltest_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@@LINUX_TRUE@v4ltest_SOURCES = v4ltest.cc
 @HAVE_X11_TRUE@@LINUX_TRUE@v4ledges_DEPENDENCIES = ../libvideogfx/libvideogfx.la
 @HAVE_X11_TRUE@@LINUX_TRUE@v4ledges_CFLAGS = $(X_CFLAGS)
 @HAVE_X11_TRUE@@LINUX_TRUE@v4ledges_LDFLAGS = -g $(X_LDFLAGS)
-@HAVE_X11_TRUE@@LINUX_TRUE@v4ledges_LDADD = ../libvideogfx/libvideogfx.la -lstdc++ $(X_LIBS) $(jpeg_libs) $(png_libs)
+@HAVE_X11_TRUE@@LINUX_TRUE@v4ledges_LDADD = ../libvideogfx/libvideogfx.la $(X_LIBS) $(jpeg_libs) $(png_libs)
 @HAVE_X11_TRUE@@LINUX_TRUE@v4ledges_SOURCES = v4ledges.cc
 all: all-am
 
diff -urN libvideogfx-1.0.8-orig/examples/symmatrix.cc libvideogfx-1.0.8/examples/symmatrix.cc
--- libvideogfx-1.0.8-orig/examples/symmatrix.cc	2013-06-24 20:45:15.000000000 +0800
+++ libvideogfx-1.0.8/examples/symmatrix.cc	2025-11-16 03:04:07.223848300 +0800
@@ -1,5 +1,5 @@
 
-#include "../libvideogfx/libvideogfx.hh"
+#include "libvideogfx/libvideogfx.hh"
 #include <iostream>
 
 using namespace videogfx;
diff -urN libvideogfx-1.0.8-orig/examples/test-libvideogfx.cc libvideogfx-1.0.8/examples/test-libvideogfx.cc
--- libvideogfx-1.0.8-orig/examples/test-libvideogfx.cc	2013-06-24 20:45:15.000000000 +0800
+++ libvideogfx-1.0.8/examples/test-libvideogfx.cc	2025-11-16 03:04:07.223848300 +0800
@@ -3,7 +3,7 @@
  */
 
 #include "config.h"
-#include "../libvideogfx/libvideogfx.hh"
+#include "libvideogfx/libvideogfx.hh"
 
 #include <iostream>
 using namespace std;
diff -urN libvideogfx-1.0.8-orig/libvideogfx/graphics/fileio/png.cc libvideogfx-1.0.8/libvideogfx/graphics/fileio/png.cc
--- libvideogfx-1.0.8-orig/libvideogfx/graphics/fileio/png.cc	2013-06-24 20:55:00.000000000 +0800
+++ libvideogfx-1.0.8/libvideogfx/graphics/fileio/png.cc	2025-11-16 03:04:07.223848300 +0800
@@ -87,8 +87,11 @@
 
   void ReadImage_PNG(Image<Pixel>& img, istream& is)
   {
+#ifdef _MSC_VER
+    assert(is);
+#else
     assert(is != NULL); // , "Open stream first.");
-
+#endif
     png_structp png_ptr;
     png_infop info_ptr;
     png_uint_32 width, height;
diff -urN libvideogfx-1.0.8-orig/libvideogfx/graphics/fileio/unified_reader.cc libvideogfx-1.0.8/libvideogfx/graphics/fileio/unified_reader.cc
--- libvideogfx-1.0.8-orig/libvideogfx/graphics/fileio/unified_reader.cc	2013-06-24 20:55:00.000000000 +0800
+++ libvideogfx-1.0.8/libvideogfx/graphics/fileio/unified_reader.cc	2025-11-20 08:57:26.102443200 +0800
@@ -488,7 +488,7 @@
     int  AskNFrames() const { return INT_MAX; }
     bool IsEOF() const { return false; }
 
-    bool SkipToImage(int nr) { id = (nr%3); }
+    bool SkipToImage(int nr) { id = (nr%3); return false; }
     void ReadImage(Image<Pixel>& img)
     {
       ImageParam spec(d_w,d_h);
diff -urN libvideogfx-1.0.8-orig/libvideogfx/types.hh libvideogfx-1.0.8/libvideogfx/types.hh
--- libvideogfx-1.0.8-orig/libvideogfx/types.hh	2013-06-24 20:45:17.000000000 +0800
+++ libvideogfx-1.0.8/libvideogfx/types.hh	2025-11-20 08:54:35.255120400 +0800
@@ -35,7 +35,7 @@
 #ifndef LIBVIDEOGFX_TYPES_HH
 #define LIBVIDEOGFX_TYPES_HH
 
-#if defined(__GCC__) && (__GNUC__ < 4) || (__GNUC__ == 4 && __GNUC_MINOR__ < 5)
+#if defined(__GCC__) && (__GNUC__ < 4) || (__GNUC__ == 4 && __GNUC_MINOR__ < 5) || defined(_MSC_VER)
 #include <stdint.h>
 #else
 #include <cstdint>
@@ -116,4 +116,12 @@
 #define NULL 0
 #endif
 
+#ifdef _WIN64
+typedef __int64 ssize_t;
+typedef __int64 pid_t;
+#else
+typedef int ssize_t;
+typedef int     pid_t;
+#endif
+
 #endif
diff -urN libvideogfx-1.0.8-orig/libvideogfx/utility/bitstream/bitbuffer.cc libvideogfx-1.0.8/libvideogfx/utility/bitstream/bitbuffer.cc
--- libvideogfx-1.0.8-orig/libvideogfx/utility/bitstream/bitbuffer.cc	2013-06-24 20:55:00.000000000 +0800
+++ libvideogfx-1.0.8/libvideogfx/utility/bitstream/bitbuffer.cc	2025-11-16 03:04:07.223848300 +0800
@@ -58,7 +58,7 @@
 
   void BitBuffer::WriteBitsMasked(uint32 bits,int nBits)
   {
-    uint32 long mask=1;
+    uint32 mask=1;
     mask<<=nBits;
     mask--;
 
